[UCSC-100] Main navigation

bjcooper opened the pull request.

Vinsanity reviewed (changes requested):
@bjcooper This is looking and functioning pretty well. A couple of changes and requests to set us up for future work.

- Let's move that nav JS to its own directory and script file. we should have separate directories in the src folder for importing these individual modules. maybe `src/js/components`. Then we can import each item in the theme.js file with the /js dir. We'll need to update the path for enqueueing as well.
- There's a z-index issue with other blocks that cover the nav on scroll.
- Can you also please update the ucsc-logo.html template part to use the new logo mentioned by Lisa. I believe I tagged you in a post somewhere in the channel.

Vinsanity commented:
@bjcooper A few items I also noticed after getting that working with ACF. 
- Can we add line-height 24px to the main nav items?
- For the nested items let's add some transition to the colors and BG there.
- For the sub nav toggle, there's not enough space on the right side. Can we increase that?
- Also on the sub nav toggle, can we drop the BG color on hover and use something else? maybe the primary yellow?

bjcooper commented:
@Vinsanity, here's what I did since your last review:

1. I added a color transition to the menu items. I applied it to the top-level menu items as well as the sub-menu items because it looked a bit weird when they acted differently.
2. I dropped the BG altogether on the sub-nav toggles since they're only shown on mobile, which doesn't really make use of :hover states.
3. I split the JS into its own file in `src/js/components` like you described.

But I have questions!

1. The line height is already 24px for mobile, but I purposely cut it down to 16px for desktop to match the designs, which do show a reduced line height. They do that because the multi-line menu items ("Admissions and Aid", for example) look terrible with the larger line height in the desktop version of the nav. So...I did not make this change. Please let me know if you're so sure that you want me to do it anyway 😄.
2. You mentioned that the sub-nav toggle doesn't have enough space on the right side, but I'm not seeing that. What I have matches their design for the mobile nav pretty closely—but it actually has 2px _more_ than they show. So...are you seeing a bug (if so, send me a screenshot) or asking for a design change (if so, let me know how much space you want)?
3. I couldn't reproduce the z-index issue you mentioned with blocks showing in front of the nav with the couple of blocks I tried. I went ahead and bumped the sticky header z-index from 1 to 10, but that's a shot in the dark without more info. If this change doesn't fix the issue, just let me know which block I can reproduce this issue with or maybe just what z-index you need it bumped to.
4. I updated the `ucsc-logo.html` template part with the new SVG you mentioned, but there's still a disconnect here. The `site-header.html` template part uses the `wp:site-logo` block and I don't see where `ucsc-logo.html` is being used at all. Please advise.

bjcooper commented:
OK, @Vinsanity, thanks for all your help on this, it should be ready for you to look again. Here's everything I changed:

I added a color transition to the menu items. I applied it to the top-level menu items as well as the sub-menu items because it looked a bit weird when they acted differently.
I dropped the BG altogether on the sub-nav toggles since they're only shown on mobile, which doesn't really make use of :hover states.
I split the JS into its own file in src/js/components like you described.
But I have questions!

- I added color hover transitions to the menu items.
- I dropped the on-hover background color for sub-menu toggles and added a transition from white to yellow on-hover (same for the menu hide/show toggle).
- I reorganized the JS as you described.
- I bumped the menu item line height for desktop up from1 6px to 18px.
- I bumped the z-index of the sticky nav up to 10.
- I updated the `ucsc-logo.html` template part with the new SVG and swapped it into the `site-header`.

Vinsanity reviewed (changes requested):
Just need a fix on that logo part and this is ready to go!

bjcooper commented:
Thank you, @Vinsanity, ready for you to look at this again!

Vinsanity reviewed (approved):
🟢 Thanks Brett! let's get this merged.
